src/docfx.website.themes/default/styles/search-worker.js

```diff
@@ -50,4 +44,35 @@
     });
     postMessage({ e: 'query-ready', q: q, d: results });
   }
+
+  function buildIndex() {
+    if (stopWords !== null && !isEmpty(searchData)) {
+      lunrIndex = lunr(function () {
+        this.pipeline.remove(lunr.stopWordFilter);
+        this.ref('href');
+        this.field('title', { boost: 50 });
+        this.field('keywords', { boost: 20 });
+
+        for (var prop in searchData) {
+          if (searchData.hasOwnProperty(prop)) {
+            this.add(searchData[prop]);
+          }
+        }
+
+        var docfxStopWordFilter = lunr.generateStopWordFilter(stopWords);
+        lunr.Pipeline.registerFunction(docfxStopWordFilter, 'docfxStopWordFilter');
+        this.pipeline.add(docfxStopWordFilter);
+        this.searchPipeline.add(docfxStopWordFilter);
+      });
+    }
+  }
+
+  function isEmpty(obj) {
+    for (var prop in obj) {
+      if (obj.hasOwnProperty(prop))
+        return false;
+    }
+
+    return JSON.stringify(obj) === JSON.stringify({});
```

Comment:
yeah, for this situation, searchData has been initialed to {}.
Or obj maybe null but it will return true(wrong judge)?

---
In reply to: [128408409](https://github.com/dotnet/docfx/pull/1891#discussion_r128408409) [](ancestors = 128408409)